scene/2d/line_builder.cpp

```diff
@@ -30,61 +30,15 @@
 
 #include "line_builder.h"
 
-//----------------------------------------------------------------------------
-// Util
-//----------------------------------------------------------------------------
-
-enum SegmentIntersectionResult {
-	SEGMENT_PARALLEL = 0,
-	SEGMENT_NO_INTERSECT = 1,
-	SEGMENT_INTERSECT = 2
-};
-
-static SegmentIntersectionResult segment_intersection(
-		Vector2 a, Vector2 b, Vector2 c, Vector2 d,
-		Vector2 *out_intersection) {
-	// http://paulbourke.net/geometry/pointlineplane/ <-- Good stuff
-	Vector2 cd = d - c;
-	Vector2 ab = b - a;
-	float div = cd.y * ab.x - cd.x * ab.y;
-
-	if (Math::abs(div) > 0.001f) {
-		float ua = (cd.x * (a.y - c.y) - cd.y * (a.x - c.x)) / div;
-		float ub = (ab.x * (a.y - c.y) - ab.y * (a.x - c.x)) / div;
-		*out_intersection = a + ua * ab;
-		if (ua >= 0.f && ua <= 1.f &&
-				ub >= 0.f && ub <= 1.f) {
-			return SEGMENT_INTERSECT;
-		}
-		return SEGMENT_NO_INTERSECT;
-	}
-
-	return SEGMENT_PARALLEL;
-}
-
-static float calculate_total_distance(const Vector<Vector2> &points) {
-	float d = 0.f;
-	for (int i = 1; i < points.size(); ++i) {
-		d += points[i].distance_to(points[i - 1]);
-	}
-	return d;
-}
-
-static inline Vector2 rotate90(const Vector2 &v) {
-	// Note: the 2D referential is X-right, Y-down
-	return Vector2(v.y, -v.x);
-}
+#include "core/math/geometry_2d.h"
 
+// Utility method
```

Comment:
```suggestion
// Utility method.
```

Reply:
I didn't do a dot because it's not a sentence, but I'm not opposed.

Reply:
idk, we seem to add periods to every comment.